kong/plugin_service.go

```diff
@@ -29,14 +31,58 @@ type AbstractPluginService interface {
 	ListAllForRoute(ctx context.Context, routeID *string) ([]*Plugin, error)
 	// Validate validates a Plugin against its schema
 	Validate(ctx context.Context, plugin *Plugin) (bool, string, error)
-	// GetSchema retrieves the schema of a plugin
+	// GetSchema retrieves the config schema of a plugin
 	GetSchema(ctx context.Context, pluginName *string) (map[string]interface{}, error)
+	// GetFullSchema retrieves the full schema of a plugin
+	GetFullSchema(ctx context.Context, pluginName *string) (map[string]interface{}, error)
+	// FillDefaults ingests plugin's defaults from its schema
+	FillDefaults(ctx context.Context, plugin *Plugin, schema map[string]interface{}) (*Plugin, error)
 }
 
 // PluginService handles Plugins in Kong.
 type PluginService service
 
-// GetSchema retrieves the schema of a plugin
+// GetFullSchema retrieves the full schema of a plugin
+func (s *PluginService) GetFullSchema(ctx context.Context,
```

Comment:
```suggestion
func (s *PluginService) GetPluginSchema(ctx context.Context,
```

Reply:
IMO this should remain as-is: it's more descriptive of the difference between this and GetSchema, and we already know this is plugin stuff since it's part of the plugin service.